### PR TITLE
IDEMPIERE-5135 : Add a link in a Broadcast message to open a record

### PR DESCRIPTION
**Pull request by nmicoud**
Credits go to devcoffee - sorry i didn't find Matheus Marcelino's github account

Co-Authored-By: muriloht <5605206+muriloht@users.noreply.github.com>

### CODE REVIEW

**Comment by nmicoud** [comment]
I've added a getUrlZoom method in MBroadcastMessage so it can be used by processes when you create the broadcast message.
eg:

String message = "bla bla bla";

MBroadcastMessage bm = new MBroadcastMessage(getCtx(), bmID, get_TrxName());
bm.setBroadcastType(MBroadcastMessage.BROADCASTTYPE_Login);
bm.setBroadcastFrequency(MBroadcastMessage.BROADCASTFREQUENCY_UntilAcknowledge);
bm.setTarget(MBroadcastMessage.TARGET_User);
bm.setAD_User_ID(recipients.get(0));
**bm.setBroadcastMessage(subject + "<br>" + message + "<br>" + bm.getUrlZoom(<your po>, "85d25fa6-72c3-4a85-9f6f-1f507851b272", "click here"));**
bm.saveEx();

**Comment by hengsin** [comment]
Hi @nmicoud ,

Example of usage should be added to the ticket instead of to the PR.

Regards,
Low

File: org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/panel/BroadcastMessageWindow.java [hunk 1]
```diff
@@ -119,6 +123,12 @@ private void init() {
 		htmlDiv.setStyle("display: table-cell; vertical-align: middle; text-align: center;");
 		Div divAlign = new Div();
 		divAlign.setStyle("color:white; position: absolute; width: 370px; height: 120px; display: table;");
+
+		Env.setContext(Env.getCtx(), MBroadcastMessage.CLIENTINFO_BROADCAST_COMPONENT_ID, pnlHead.getUuid());
+		textMsgContent.setContent(mbMessages.get(0).get_Translation(MBroadcastMessage.COLUMNNAME_BroadcastMessage));
+		pnlHead.addEventListener(ZoomEvent.EVENT_NAME, this);
+		htmlDiv.setFocus(true);
```
Comment (hengsin) [inline org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/panel/BroadcastMessageWindow.java:130]: this added block should replace "textMsgContent.setContent(mbMessages.get(0).get_Translation(MBroadcastMessage.COLUMNNAME_BroadcastMessage));" at line 117.

Comment (nmicoud) [reply org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/panel/BroadcastMessageWindow.java]: done

**Review by hengsin — APPROVED** [review]
looks good

**Pull request merged**